agent/ec2/ec2_client.go

```diff
@@ -87,10 +90,14 @@ func (c *ClientImpl) DescribeECSTagsForInstance(instanceID string) ([]*ecs.Tag,
 	var tags []*ecs.Tag
 	// Convert ec2 tags to ecs tags
 	for _, ec2Tag := range res.Tags {
-		tags = append(tags, &ecs.Tag{
-			Key:   ec2Tag.Key,
-			Value: ec2Tag.Value,
-		})
+		// Filter out all tags "aws:" prefix
+		if !strings.HasPrefix(strings.ToLower(aws.StringValue(ec2Tag.Key)), awsTagPrefix) &&
+			!strings.HasPrefix(strings.ToLower(aws.StringValue(ec2Tag.Value)), awsTagPrefix) {
```

Comment:
even values with aws prefix are rejected?

Reply:
The service will still reject something with key "aws:<foo>" and value "<bar>", right? We should change this if so.

Reply:
Yes, value with aws prefix will be rejected as well, see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions

Tags with key "aws:foo" and value "bar" will be filtered out here, I added test cases.